autoload/phpactor/project/assigner.vim

```diff
@@ -1,9 +1,9 @@
-function! phpactor#project#assigner#assignProjectToBuffer(filename)
+function! phpactor#project#assigner#assignProjectToBuffer(filename, allowReassign)
   if !exists('g:phpactorLoaded')
     return
   endif
 
-  if exists('b:project')
+  if exists('b:project') && (v:false is a:allowReassign)
```

Comment:
How can a buffer need to change project ?
If it was moved then is it still the same buffer ?
`filename`  will be different I think, not sure about what I'm saying so it needs to be checked.
If a file can be moved on the filesystem without changing buffer then we might want to detect that instead of providing a flag. Maybe stocking the filename in buffer variable and checking it against `a:filename`.

Talking about buffer variable make me wonder if we would not need a more unique name than `b:buffer`, it's a shared namespace so we should include `phpactor` in it at least. To limit the risk of collisions with other plugin.

Reply:
I agree that this variable should be removed to something containing "namespace". 

At the moment the only reason to change is [a choice hidden from the user](https://github.com/phpactor/phpactor/pull/959#issuecomment-615840400): `inputlist()` non-interactively returns its default value so there is need to change it later.

Reply:
I didn't saw your previous explanation, sorry.

So for me the second option is a no go, we can't ask people to configure their editor so that they can use our tool, it's against the purpose IMO.

For the first option, as long as the default value used is configurable I don't see the problem.
And as I said I didn't do much except reading your code until know.
Stupidly just to see what was going on without try to understand how it will work.
One reason for that is that I never thought the user will be impacted.
If we need to handle some kind of project files grouping to be able to do our work: fine.
But the user should not be bother with that outside of configuration.
Then I start to think that what you did will ask the user in which project the file should go ?
I hope I mistaken because:
```gherkin
As a user
Given I use a tool to improve my editor
When this tool makes my editor ask me a question every time I open a file
Then I burn down the tool and send it to hell with a voodoo dance!
```

Reply:
Sadly, I can be not able to recognize the boundaries of irony in your reply ;) My English is such as you see and you can expect more misunderstandings ;)

I would like to allow provide the prompt for each new buffer that does not belong to any project. It is far from prompting for each file, its rather a rare case per the whole session. Right, probably we should allow disable this dialog completely and configure phpactor to use one or more non-interactive methods or  (as currently or in a configurable way, e.g. [`initialCwd`, `rootMarkers`]). 

As we discussed earlier, each method has downsides so allowing to interactive choose one of them is a good idea. 

I have no idea how to simulate user input into dialog with vader test but I in such case I we can do it manually and allow to continue automatically, as usual. Such test would be not for run by CI, but maybe it can help to understand what can happen from the user point of view.

Reply:
> So for me the second option is a no go, we can't ask people to configure their editor so that they can use our tool, it's against the purpose IMO.

Only possibility to interactive choose the method depending (at the moment) on a presence `F` in `shortmess`. With `F` only the interactive choice is unavailable - not the whole phpactor. 

It would be good to minimize dependency of user vim settings. Notice that other tools can also rely on some of them. For example, [set hidden is suggested for using CoC](https://github.com/neoclide/coc.nvim/wiki/F.A.Q#my-vim-is-blocked-sometimes).